### ISSUE TITLE
TypeError when CTCData compress=True

### ISSUE THREAD

**Issue by kevinyamauchi** [weigertlab/trackastra#20]
Hello! I was trying to use the `compress` option for the `CTCData` dataset class and I received the `TypeError` below. These are 3D images and they work with `compress` set to `False`. 

We can work with smaller training datasets for now, so this isn't a total blocker for us - just wanted to flag in case there's an easy fix.

```python
❯ python train_trackastra.py -c track_astra_config_full_simulation.yaml --compress True
WARNING:__main__:Training from scratch, this is slow!

INFO:__main__:Load one dataset to preallocate memory for training
INFO:trackastra.data.data:ROOT (config): track_data/simulation_1_voxel_size_075_um
INFO:trackastra.data.data:ROOT: 	track_data/simulation_1_voxel_size_075_um
INFO:trackastra.data.data:GT TRA:	track_data/simulation_1_voxel_size_075_um_GT/TRA
INFO:trackastra.data.data:GT MASK:	track_data/simulation_1_voxel_size_075_um_GT/TRA
INFO:trackastra.data.data:IMG:		track_data/simulation_1_voxel_size_075_um
INFO:trackastra.data.data:Loading images                                        
Traceback (most recent call last):                                              
  File "/local1/code/mechanomorph/scripts/train_trackastra.py", line 1277, in <module>
    vars = train(args)
  File "/local1/code/mechanomorph/scripts/train_trackastra.py", line 793, in train
    dummy_data = CTCData(
  File "/local0/miniconda/envs/trackastra/lib/python3.10/site-packages/trackastra/data/data.py", line 221, in __init__
    self.windows = self._load_wrfeat()
  File "/local0/miniconda/envs/trackastra/lib/python3.10/site-packages/trackastra/data/data.py", line 1028, in _load_wrfeat
    self.imgs = np.stack(
  File "/local0/miniconda/envs/trackastra/lib/python3.10/site-packages/numpy/core/shape_base.py", line 364, in _stack_dispatcher
    arrays = _arrays_for_stack_dispatcher(arrays)
  File "/local0/miniconda/envs/trackastra/lib/python3.10/site-packages/numpy/core/shape_base.py", line 209, in _arrays_for_stack_dispatcher
    raise TypeError('arrays to stack must be passed as a "sequence" type '
TypeError: arrays to stack must be passed as a "sequence" type such as list or tuple.

```

**Comment by bentaculum** [comment]
Hi @kevinyamauchi,
happy to look into this. Would you mind sharing your full config please?

One guess I have is that the reader in the `CTCData` somehow gets confused, since it tries handling multiple data formats.
The current code accepts either our simplified CTC data format:

```
<dataset_root_in_config>
-- img
-- TRA
```

or original CTC data format:
```
<parent>
-- <dataset_root_in_config>    (images)
-- <dataset_root_in_config>_GT/TRA    (tracking ground truth, might contain points only)
-- <dataset_root_in_config>_ST/SEG    (dense masks)
```

**Comment by kevinyamauchi** [comment]
Thanks for looking into it, @bentaculum !

To reproduce, you can do the following:
1. download the Fluo-N2DL-HeLa folder from the README
2. Use the train_trackastra.py script from the `scripts` folder (config below):  `python train_trackastra.py --config track_astra_demo.yaml --compress True`
3. I obtain the error below

As I mentioned, this isn't totally blocking, so no big pressure from my side. Thanks!

## Config
from the `scripts` folder example - just modified the path to the data
```yaml
batch_size: 8
crop_size:
- 256
- 256
detection_folders:
- TRA
dropout: 0.01
example_images: False  # Slow
input_train:
- Fluo-N2DL-HeLa/02
input_val:
- Fluo-N2DL-HeLa/01
max_tokens: 2048
name: compress_example
ndim: 2
num_decoder_layers: 5
num_encoder_layers: 5
outdir: runs
distributed: False
window: 6
```

## Traceback
```python
❯ python train_trackastra.py --config track_astra_demo.yaml --compress True
WARNING:__main__:Training from scratch, this is slow!

INFO:__main__:Load one dataset to preallocate memory for training
INFO:trackastra.data.data:ROOT (config): Fluo-N2DL-HeLa/02
INFO:trackastra.data.data:ROOT: 	Fluo-N2DL-HeLa/02
INFO:trackastra.data.data:GT TRA:	Fluo-N2DL-HeLa/02_GT/TRA
INFO:trackastra.data.data:GT MASK:	Fluo-N2DL-HeLa/02_GT/TRA
INFO:trackastra.data.data:IMG:		Fluo-N2DL-HeLa/02
INFO:trackastra.data.data:GT MASK:      Fluo-N2DL-HeLa/02_ST/SEG for correcting with ST masks                      
INFO:trackastra.data.data:Loading images                                                                           
Traceback (most recent call last):                                                                                 
  File "/local1/code/mechanomorph/scripts/train_trackastra.py", line 1277, in <module>
    vars = train(args)
  File "/local1/code/mechanomorph/scripts/train_trackastra.py", line 793, in train
    dummy_data = CTCData(
  File "/local0/miniconda/envs/trackastra/lib/python3.10/site-packages/trackastra/data/data.py", line 221, in __init__
    self.windows = self._load_wrfeat()
  File "/local0/miniconda/envs/trackastra/lib/python3.10/site-packages/trackastra/data/data.py", line 1028, in _load_wrfeat
    self.imgs = np.stack(
  File "/local0/miniconda/envs/trackastra/lib/python3.10/site-packages/numpy/core/shape_base.py", line 364, in _stack_dispatcher
    arrays = _arrays_for_stack_dispatcher(arrays)
  File "/local0/miniconda/envs/trackastra/lib/python3.10/site-packages/numpy/core/shape_base.py", line 209, in _arrays_for_stack_dispatcher
    raise TypeError('arrays to stack must be passed as a "sequence" type '
TypeError: arrays to stack must be passed as a "sequence" type such as list or tuple.
```

**Comment by bentaculum** [comment]
Thanks for flagging this bug! Now fixed in b6d231e, and releasing atm.

**Comment by kevinyamauchi** [comment]
@bentaculum, thank you for the quick fix! The compress flag now works for me.
